### ISSUE TITLE
x-html + x-model causes undefined data error

### ISSUE THREAD

**Issue by dubcanada** [alpinejs/alpine#766]
Hello,

I have a very complex example where I am attempting to build a tree with checkboxes, everything is good (minus some performance issues with a large number of nodes).

I have replicated my error down to a smaller jsfiddle

https://jsfiddle.net/wtojnhvy/

Basically the issue is in the x-html I have a x-model which requires a value of the currently open group passed in `groups[group_id].files[file.path]`

This works perfectly fine until I check the checkbox, I then get an error that group_id is undefined and it re-renders the component causing the currently open leaf to close.

Basically go to that jsfiddle, click `test` and then check off test 2 or test 3.

Obviously this is someone that would be solved with v3, but for now is there anyway to solve this issue?

**Comment by calebporzio** [comment]
Hey @dubcanada, thanks for providing a Fiddle for the issue.

I haven't looked into it deeply as I don't believe Alpine will support dynamic syntax inside of `x-html`. I don't know if there is a precedence for this in Vue, but in my mind `x-html` is for non-reactive bits of HTML.

So before we go further, is there a non-`x-html` oriented strategy you could use to achieve this same result?

Thanks!

**Comment by dubcanada** [comment]
Not without recursive components sadly since the tree depth is unknown. v3 will allow such a thing with alpine components, but I can't think of a method using v2.

**Comment by KevinBatdorf** [comment]
@dubcanada `x-html` will re-render on every update so you need to manage the state from outside of that function. 

You can pass in the properties you need to be updated though and it will work, recursively even. I do this in a "dev tools" component I'm working on. Maybe you can pul some ideas from there:

https://github.com/KevinBatdorf/alpine-inline-devtools/blob/master/index.js#L124

**Comment by dubcanada** [comment]
@KevinBatdorf thank you very much for your help, that was the issue.

However it does mean I cannot do it this way, as rendering a tree every time a model changes is not going to work :(
